p2p/host/autonat/svc.go

```diff
@@ -273,6 +273,11 @@ func (as *autoNATService) Disable() {
 	}
 }
 
+func (as *autoNATService) Close() error {
+	as.Disable()
+	return as.config.dialer.Close()
+}
+
```

Comment:
Should we just close the dialer from the autonat instance? That seems better, we ensure that the dialer is always closed even if for some reason there's an issue with service being set to nil for some reason. 
I don't have strong opinions here.

Reply:
My read here is that the dialer is owned by the service, so it's the service's responsibility to close it.